src/Pages/VideoListing/VideoCard.jsx

```diff
@@ -0,0 +1,48 @@
+import { useUserContext } from "../../Context";
+import { updateLikedVideos, updateWatchLater } from "../../Utils";
+import { Link } from "react-router-dom";
+
+export function VideoCard({ video }) {
+  const { title, creator, _id, category } = video;
+  const thumbnail = `https://i.ytimg.com/vi/${_id}/maxresdefault.jpg`;
+
+  const {
+    userState: { likedVideos, watchLater },
+    userDispatch,
+  } = useUserContext();
+
+  const isVideoInLikeList = likedVideos.some((vid) => vid._id === video._id);
+  const isVideoInWatchLater = watchLater.some((vid) => vid._id === video._id);
+  return (
+    <div className="card video-card txt-left">
+      <Link to={`/videos/${_id}`} className="link-no-style">
+        <img className="card-img" src={thumbnail} alt={category} />
+        <p className="heading-5 l-sp-1 txt-white txt-bold pd-sm-left ee">
+          {title.length > 25 ? `${title.slice(0, 25)}...` : title}
```

Comment:
For long titles instead of doing this, you can simply use the ellipsis property of text-overflow in CSS. it will handle this automatically. Read about it.
you can use this css
For long titles instead of doing this, you can simply use the ellipsis property of text-overflow in CSS. it will handle this automatically. Read about it.
so you have to not calculate tile length

Reply:
I only want title upto 25 characters so didn't use ellipsis